### ISSUE TITLE
Renames

### ISSUE THREAD

**Issue by HalfVoxel** [cmancone/mygrations#1]
Hi

I was wondering if it is possible to rename columns or tables using this tool?
It seems like a not too uncommon thing to do during a migration, but I don't see how it fits with the architecture of this library.

**Comment by cmancone** [comment]
@HalfVoxel This item is actually on my list.  You're right - as is this system can't really support such a concept.  However, as I have been thinking about it, I think there is an easy solution anyway (I hope).  The system already supports SQL comments inside the files.  I have been thinking about the ability to put commands for the mygration system inside SQL comments inside the file (or perhaps coming up with a command syntax that is mygrations specific, but I like the idea of having it inside SQL comments so that the files themselves remain valid SQL).

Thinking of something like:

```
/* @RenameTable old_table_name new_table_name */
CREATE TABLE new_table_name (...);
```

OR

```
/* @RenameColumn table_name.old_column_name table_name.new_column_name */
CREATE TABLE table_name (...);
```
That's just the first thought off the top of my head.  I'm definitely up for suggestions.  I'm sure there would be more uses for such a system over time, so I would want to think this out.

**Comment by HalfVoxel** [comment]
I would personally do it as some kind of alias comment on the column.
Like

~~~
`price` decimal(15,2) NOT NULL, /* @previously_named(cost) */
~~~

**Comment by cmancone** [comment]
The only challenge to watch out for there is to make sure there is no ambiguity about which column was renamed, i.e. which column is being renamed here:

```
`price` decimal(15,2) NOT NULL,
/* @previously_named(cost) */
`total` decimal(15,2) NOT NULL
```
With a comment on the end of the line you are implicitly applying the rename to the column on the same line.  However from a language syntax perspective there isn't any real difference between your example and having the rename comment on a line by itself, and then the column you are asking to rename is more ambiguous.

I would probably either make a standard rule (the rename command applies to the column that immediately follows it) or put the old name and new name explicitly in the rename command to avoid confusion:

```price` decimal(15,2) NOT NULL, /* @renamed(cost, price) */```

I suppose you could always do both and make the `price` (in my latest example) optional, understanding that if it is not provided the rename applies to `SOME FIXED RULE` column.

**Comment by HalfVoxel** [comment]
That is true. I'm not sure if you have your own custom parser or not, but if you do you could perhaps add it in a way that is only allowed in a single position in the definition.

like
~~~
`price` decimal(15,2) NOT NULL MYGRATIONS_PREVIOUS_NAME(cost)
~~~
or something.
